[8.3.4] supply-chain CI job is silently failing on every PR: advisory-db cache missing

## Symptom

`supply-chain` (cargo-deny via `EmbarkStudios/cargo-deny-action@v2.0.17`) has been failing on every PR since at least 2026-04-23. Errors consistently with:

\`\`\`
[ERROR] failed to get 'FETCH_HEAD' metadata:
  failed to parse ISO-8601 timestamp 'fatal: cannot change to
    '/home/runner/.cargo/advisory-dbs/advisory-db-3157b0e258782691':
    No such file or directory': failed to parse year in date:
    failed to parse four digit integer as year: invalid digit,
    expected 0-9 but got f
\`\`\`

Confirmed on PR #535 both on the initial run (job `72735191634`, 43s) and on `--failed` rerun (job `72735770763`, 33s) — same error, same timestamp-parsing confusion over the missing advisory-db directory.

## Impact

**Silent.** The job has `continue-on-error: true` in `.github/workflows/ci.yml`, so `ci-success` (the required aggregator) stays green and merges aren't blocked. But:

1. We have **zero supply-chain / advisory coverage** right now — every RUSTSEC advisory against our deps is going undetected.
2. Every PR has a red ✗ next to `supply-chain`; a reviewer glancing at checks has to re-learn each time that it's non-blocking by design.
3. If a real supply-chain failure (a CVE in a dep we actually use) lands, it will be indistinguishable from the current infra noise and we may wave it through.

## Root cause candidates

1. `EmbarkStudios/cargo-deny-action@v2.0.17` is pinned to SHA `91bf2b620e09e18d6eb78b92e7861937469acedb` and may have a stale advisory-db fetch path. A newer release may fix the cache init.
2. GitHub Actions runner image change broke the `~/.cargo/advisory-dbs/` directory layout the action expects.
3. Rustsec advisory-db repo schema changed in a way the pinned action can't parse.

## Proposed fix

Three paths, pick in order of least-churn:

1. **Bump the action pin** — update `EmbarkStudios/cargo-deny-action` from `v2.0.17` to the latest. Verify on a sacrificial PR.
2. **Pre-create the advisory-db directory** in a previous step: `mkdir -p ~/.cargo/advisory-dbs` before the action runs. Hacky but might work.
3. **Replace with `rustsec/audit-check`** which has different advisory-db fetch semantics.

## Acceptance

- [ ] `supply-chain` job shows a green ✓ on a fresh PR.
- [ ] A deliberately introduced test advisory (e.g. pull in a known-vulnerable dep version as a no-op dev-dep) causes the job to fail with a recognizable error — i.e. it's actually auditing our deps again.
- [ ] Remove `continue-on-error: true` once the job is stable, so real supply-chain failures block merge.

Surfaced during: PR #535 CI run (job `72735191634`).

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -227,11 +227,17 @@ jobs:
       - name: Install Rust toolchain
         uses: dtolnay/rust-toolchain@stable
 
+      # #536: EmbarkStudios/cargo-deny-action@v2.0.17 was failing on every
+      # PR with `failed to get 'FETCH_HEAD' metadata` — the action's internal
+      # gix-based advisory-db cache didn't exist on fresh runners and
+      # couldn't be bootstrapped. Install cargo-deny directly via cargo
+      # instead; its own advisory-db fetch handles the `~/.cargo/advisory-db`
+      # path correctly (same path cargo-audit uses in the rust-checks job).
+      - name: Install cargo-deny
+        run: cargo install cargo-deny --locked
+
       - name: Run cargo-deny
-        uses: EmbarkStudios/cargo-deny-action@91bf2b620e09e18d6eb78b92e7861937469acedb # v2.0.17
-        with:
-          command: check
-          arguments: --all-features
+        run: cargo deny --all-features check
 
   ci-success:
     if: always()

